src/convenience.jl

```diff
@@ -0,0 +1,186 @@
+end
+
+"""
+    stack(filename; kw...) => AbstractGeoStack
+    stack(filenames::NamedTuple; kw...) => AbstractGeoStack
+
+Load a file path or a `NamedTuple` of paths as an `AbstractGeoStack`.
+
+# Arguments
+
+- `filename`: A `NamedTuple` of stack keys and `String` filenames.
+
+# Keywords
+
+Passed to the constructor for the file type, and commmonly include:
+
+- `window`: A `Tuple` of `Dimension`/`Selector`/indices that will be applied to the
+    contained arrays when they are accessed.
+- `metadata`: Metadata as a `StackMetadata` object.
+- `child_kwargs`: A `NamedTuple` of keyword arguments to pass to the `childtype` constructor.
+- `refdims`: `Tuple` of  position `Dimension` the array was sliced from.
+
+```julia
+files = (:temp="temp.tif", :pressure="pressure.tif", :relhum="relhum.tif")
+stack = stack(files; child_kwargs=(mappedcrs=EPSG(4326),))
+stack[:relhum][Lat(Contains(-37), Lon(Contains(144))
+```
+"""
+function stack end
+
+stack(filename::AbstractString; kw...) = _constructor(stack, filename)(filename; kw...)
+function stack(filenames::NamedTuple; child_kwargs=(), kw...)
+    if all(x -> splitext(x)[2] == splitext(first(filenames))[2], filenames)
+        # All the same kind of file. We don't need to load them up front.
+        DiskStack(filenames; childtype=_constructor(geoarray, first(filenames)), kw...)
+    else
+        # These files are different extensions, just load them all
+        # as separate `AbstarctGeoArray` (which has some up front cost from
+        # reading the dimensions). They are probably still disk-backed for the
+        # actual array.
+        arrays = map(filenames) do fn
+            _constructor(geoarray, fn)(fn; child_kwargs...)
+        end
+        GeoStack(arrays; kw...)
+    end
+end
+
+"""
+    Base.write(filename::AbstractString, T::Type{<:AbstractGeoArray}, s::AbstractGeoStack)
+
+Write any [`AbstractGeoStack`](@ref) to file, guessing the backend from the file extension.
+
+Keyword arguments are passed to the `write` method for the backend.
+
+If the source can't be save as a stack-like object, individual array layers will be saved.
+"""
+function Base.write(filename::AbstractString, s::AbstractGeoStack; kw...)
+    base, ext = splitext(filename)
+    T = _constructor(stack, filename; throw=false)
+    if T isa Nothing
+        # Save as individual `geoarray`
+        T = _constructor(geoarray, filename)
+        for key in keys(s)
+            fn = joinpath(string(base, "_", key, ext))
+            write(fn, T, s[key])
+        end
+    else
+        write(filename, _constructor(stack, filename), s; kw...)
+    end
+end
+
+"""
+    series(dirpath::AbstractString, dims; kw...) => AbstractGeoSeries
+
+Load a vector of filepaths as a `AbstractGeoSeries`. `kw` are passed to the constructor.
+
+`dims` Dimensions can hold an index matching the files in the directory,
+or a function to convert the filename strings to index values.
+"""
+function series end
+
+function series(dirnames; 
```

Comment:
I actually never used `series` nor am I really familiar with what their usecase is.  Anyway, this function returns not a GeoSeries but a GeoStack.  Is this intentional?  If so, is it not contradicting the doc-string?

Reply:
Oh right good catch, I'm not sure what I've done here. It should return a series.

Series is useful for lazy loading huge datasets that are split in multiple files over some dimension/s, often time. So you can index it with the time and get a stack/geoarray result.